docs/community/contribute.md

```diff
@@ -4,21 +4,114 @@ MXNet has been developed and is used by a group of active community members.
-* [Document](#documents)
-* [Test Cases](#test-cases)
-* [Examples](#examples)
-* [Tutorials](#tutorials)
-* [Core Library](#core-library)
-* [Python Package](#python-package)
-* [R Package](#r-package)
-
-### Submitting a Pull Request
+## Code Contribution
+
+### Core Library
+
+- Follow the Google C++ Style Guide for C++ code.
+- Use doxygen to document all of the interface code.
+- To reproduce the linter checks, type ```make lint```.
+
+### Python Package
+
+- Always add docstring to the new functions in numpydoc format.
+- To reproduce the linter checks, type ```make lint```.
+
+### R Package
+
+#### Code Style
+- Most of the C++ code in the R package relies heavily on [Rcpp](https://github.com/RcppCore/Rcpp).
+- We follow the Google C++ Style Guide for C++ code. This allows us to maintain consistency with the rest of the project. It also allows us to check style automatically with a linter.
+- To check the code style, type the following command at the root folder:
+```bash
+make rcpplint
+```
+- If necessary, disable the linter warning on certain lines with ```// NOLINT(*)``` comments.
+
+#### Auto-Generated API
+- Many MXNet APIs are exposed dynamically from Rcpp.
+- mxnet_generated.R is the auto-generated API and documents for these functions.
+- Remake the file by typing the following command at root folder:
+```bash
+make rcppexport
+```
+- Use this command only when there is an update to dynamic functions.
+
+#### API Document
+The document is generated using roxygen2. To remake the documents in the root folder, use the following command:
+```bash
+make roxygen.
+```
+
+#### R Markdown Vignettes
+R Markdown vignettes are located on GitHub in [R-package/vignettes](https://github.com/dmlc/mxnet/tree/master/R-package/vignettes).
+These R Markdown files aren't compiled. We host the compiled version on [doc/R-package](https://github.com/dmlc/mxnet/tree/master/R-package/).
+
+To add a new R Markdown vignettes:
+
+* Add the original R Markdown file to ```R-package/vignettes```
+* Modify ```doc/R-package/Makefile``` to add the Markdown files to be built.
+* Clone the [dmlc/web-data](https://github.com/dmlc/web-data) repo to  the  ```doc``` folder.
+* Type the following command for the ```doc/R-package``` package:
+```bash
+make the-markdown-to-make.md
+```
+* This generates the markdown, and the figures and places them into ```doc/web-data/mxnet/knitr```.
+* Modify the ```doc/R-package/index.md``` to point to the generated markdown.
+* Add the generated figure to the ```dmlc/web-data``` repo.
+	* If you have already cloned the repo to doc, use ```git add```.
+* Create a pull request for both the markdown  and ```dmlc/web-data```.
+* You can also build the document locally with the following command: ```doc```
+```bash
+make html
+```
+This prevents radically increasing the size of the repo with generated images sizes.
+
+### Test Cases
+
+* All of the test cases are in GitHub in [tests](https://github.com/dmlc/mxnet/tree/master/tests)
```

Comment:
"All of the test cases are in GitHub in" -> "All of our tests can be found in the Github repo in the following directory: "